Fixes libraries doesnt get downloaded into sites/all/libraries.

leolandotan opened the pull request:
This issue seems to arise as composer updated it's versions and
maybe updated their code specifically in LibraryInstaller. I also
noticed that the getPackageBasePath() doesn't get called instead
the getInstallPath() is being called by adding a simple
var_dump("test"); into the functions during composer install.

I haven't got the root cause for this but this.

leolandotan committed:
Fixes libraries doesnt get downloaded into sites/all/libraries.

This issue seems to arise as composer updated it's versions and
maybe updated their code specifically in LibraryInstaller. I also
noticed that the getPackageBasePath() doesn't get called instead
the getInstallPath() is being called by adding a simple
var_dump("test"); into the functions during composer install.

I haven't got the root cause for this but this but I got the idea
from https://getcomposer.org/doc/articles/custom-installers.md.

generalredneck commented on src/Drupal/Composer/DrupalLibraryInstaller.php:
```diff
@@ -37,4 +37,17 @@ public function getPackageBasePath(PackageInterface $package)
+        $path = $this->drupalLibrariesPath . $this->drupalLibraryMap[$package->getPrettyName()];
+      }
+      return $path;
+    }
```
DRY
Don't Repeat Yourself

Copy and paste is not allowed.

leolandotan replied:
Noted. Will update it.

generalredneck commented:
FYI https://github.com/composer/composer/issues/4648

leolandotan committed:
Fix code to use best practices and remove unused function.

The reference why the old function was removed is because of this
reported issue https://github.com/composer/composer/issues/4648.
